crates/topos-sequencer-subnet-runtime/src/certification.rs

```diff
@@ -126,7 +131,27 @@ impl Certification {
         }
 
         // Remove processed blocks
-        self.finalized_blocks.clear();
+        for processed_block_number in processed_blocks {
```

Comment:
I don't understand this code block. When I read the code above I see that every `block_number` are added to `processed_blocks`, if an error happens then this code block isn't even executed. Am I missing something?

Reply:
Yes, we only want to remove block from `self.finalized_blocks` if the certificate is generated for that block. Otherwise, it will be processed again on the next certificate call.  `processed_blocks` keeps the blocks for which the certificate is generated

Reply:
Wouldn't `certified_blocks` be a better name than `processed_blocks` then?